### PR TITLE
Refactor Ballerina Code to Avoid Anti-patterns

### PR DESCRIPTION
**Pull request by ThisaruGuruge**
## Purpose
$Subject

Fixes: [#2244](https://github.com/ballerina-platform/ballerina-standard-library/issues/2244)

## Checklist
- [x] Linked to an issue
- [ ] ~Updated the changelog~
- [ ] ~Added tests~

### CODE REVIEW

File: ballerina/listener_utils.bal [hunk 1]
```diff
@@ -332,7 +332,7 @@ isolated function getHttpService(Engine gqlEngine, GraphqlServiceConfig? service
             if context is error {
                 json payload = { errors: [{ message: context.message() }] };
```
Comment (DimuthuMadushan) [inline ballerina/listener_utils.bal:333]:
```suggestion
                json payload = {errors: [{message: context.message()}]};
```

File: ballerina/modules/parser/utils.bal [hunk 2]
```diff
@@ -49,31 +49,28 @@ isolated function getAnonymousOperationInMultipleOperationsError(OperationNode o
...
     TokenType kind = token.kind;
-    if (kind == T_EOF) {
+    if kind == T_EOF {
         return "<EOF>";
-    } else if (kind == T_IDENTIFIER) {
+    } else if kind == T_IDENTIFIER {
         return getScalarTypeNameForError(token.value);
-    } else if (kind == T_STRING) {
+    } else if kind == T_STRING {
         return string`String "${token.value}"`;
```
Comment (DimuthuMadushan) [inline ballerina/modules/parser/utils.bal:71]:
```suggestion
        return string `String "${token.value}"`;
```

File: ballerina/modules/parser/utils.bal [hunk 3]
```diff
@@ -49,31 +49,28 @@ isolated function getAnonymousOperationInMultipleOperationsError(OperationNode o
...
-    } else {
-        return string`"${token.value}"`;
     }
+    return string`"${token.value}"`;
```
Comment (DimuthuMadushan) [inline ballerina/modules/parser/utils.bal:73]:
```suggestion
    return string `"${token.value}"`;
```

File: ballerina/tests/test_services.bal [hunk 4]
```diff
@@ -146,14 +150,14 @@ service /inputs on basicListener {
...
 
     isolated resource function get isHoliday(Weekday? weekday) returns boolean {
-        if (weekday == SUNDAY || weekday == SATURDAY) {
+        if weekday == SUNDAY || weekday == SATURDAY {
             return true;
         }
         return false;
     }
 
     isolated resource function get getDay(boolean isHoliday) returns Weekday[]{
```
Comment (DimuthuMadushan) [inline ballerina/tests/test_services.bal:159]:
```suggestion
    isolated resource function get getDay(boolean isHoliday) returns Weekday[] {
```

File: ballerina/tests/test_services.bal [hunk 5]
```diff
@@ -389,25 +393,25 @@ service /service_objects on serviceTypeListener {
...
             return p1;
         } else {
             return a1;
         }
     }
 
     resource function get details(int id) returns Details {
-        if (id < 5) {
+        if id < 5 {
             return { information: p1 };
```
Comment (DimuthuMadushan) [inline ballerina/tests/test_services.bal:415]:
```suggestion
            return {information: p1};
```

File: ballerina/tests/test_services.bal [hunk 6]
```diff
@@ -389,25 +393,25 @@ service /service_objects on serviceTypeListener {
...
         } else {
             return { information: a1 };
```
Comment (DimuthuMadushan) [inline ballerina/tests/test_services.bal:417]:
```suggestion
            return {information: a1};
```

File: ballerina/validator_visitor.bal [hunk 7]
```diff
@@ -150,7 +150,7 @@ class ValidatorVisitor {
                     }
                 }
                 if !isProvidedField {
-                    if ((subInputValue.'type).kind == NON_NULL && schemaArg?.defaultValue is ()) {
+                    if subInputValue.'type.kind == NON_NULL && schemaArg?.defaultValue is () {
                         string inputFieldName = getInputObjectFieldFormPath(self.argumentPath, subInputValue.name);
                         string message = string`Field "${inputFieldName}" of required type ` +
```
Comment (DimuthuMadushan) [inline ballerina/validator_visitor.bal:155]:
```suggestion
                        string message = string `Field "${inputFieldName}" of required type ` +
```

File: ballerina/validator_visitor.bal [hunk 8]
```diff
@@ -150,7 +150,7 @@ class ValidatorVisitor {
...
                         string`"${getTypeNameFromType(subInputValue.'type)}" was not provided.`;
```
Comment (DimuthuMadushan) [inline ballerina/validator_visitor.bal:156]:
```suggestion
                        string `"${getTypeNameFromType(subInputValue.'type)}" was not provided.`;
```

Comment (DimuthuMadushan) [reply ballerina/validator_visitor.bal:156]: There are a few other places with the same issue.

File: ballerina/modules/parser/tests/parser_tests.bal [hunk 9]
```diff
@@ -961,11 +961,11 @@ isolated function testListTypeArgumentsInDirecitves() returns error? {
 }
 isolated function testVariablesWithInvalidDefaultValue() returns error? {
     string document = "query getId($name: String = $name) { profile(userName:$name) { id } }";
```
Comment (DimuthuMadushan) [inline ballerina/modules/parser/tests/parser_tests.bal:963]:
```suggestion
    string document = "query getId($name:String = $name) { profile(userName:$name) { id } }";
```

File: ballerina/modules/parser/tests/parser_tests.bal [hunk 10]
```diff
@@ -533,20 +533,20 @@ isolated function testVariables() returns error? {
 }
 isolated function testNonNullTypeVariables() returns error? {
     string document = "query getId($name: String!, $age: Int!) { profile(userName:$name, userAge:$age) { id } }";
```
Comment (DimuthuMadushan) [inline ballerina/modules/parser/tests/parser_tests.bal:535]:
```suggestion
    string document = "query getId($name:String!, $age:Int!) { profile(userName:$name, userAge:$age) { id } }";
```

Comment (ThisaruGuruge) [reply ballerina/modules/parser/tests/parser_tests.bal:535]: Is this really needed? There is a space before the Type in GraphQL documents, right?

File: ballerina/modules/parser/tests/parser_tests.bal (hunk 9 above)
Comment (ThisaruGuruge) [reply ballerina/modules/parser/tests/parser_tests.bal:963]: Ditto

**Review by DimuthuMadushan — APPROVED** [review]

**Review by shafreenAnfar — APPROVED** [review]

**Pull request merged**